pandas/core/arrays/datetimes.py

```diff
@@ -405,7 +394,13 @@ def _generate_range(
                 if end is not None:
                     end = end.tz_localize(None)
 
-            values, _tz = generate_regular_range(start, end, periods, freq)
+            if isinstance(freq, Tick):
+                values = generate_regular_range(start, end, periods, freq)
+            else:
+                xdr = generate_range(start=start, end=end, periods=periods, offset=freq)
```

Comment:
is this some kind of list returned?

Reply:
`generate_range` is an iterator function present in `pandas/tseries/offsets.py`.  So, here, `xdr` is an iterator (hence the conversion in a numpy array just after). This code was hidden in the previous version of `generate_regular_range`.

---
To be complete, in an earlier version of this PR I created two wrappers of `generate_regular_range`:

* `generate_timestamps_range`: wrapper of `generate_regular_range`, and used to call `generate_range` to handle the cases when freq was not an instance of `Tick`
* `generate_timedeltas_range`: wrapper of `generate_regular_range`

Both of them were really small and a bit unnecessary (especially `generate_timedeltas_range`). So it has been removed after a remark of @jbrockmendel https://github.com/pandas-dev/pandas/pull/33498#discussion_r408425826.